Limit 0 not allowed for redrive to queue:

carioca-au opened the pull request:
- Limit 0 or lower, now raises an exception informing that value isn't valid.
- None is the default value, which will redrive the whole queue
- get_messages was changed to not allow 0 as limit and raise an exception in case of limit lower than 1.

Commit:
Limit 0 not allowed for redrive to queue:

- Limit 0 or lower, now raises an exception informing that value isn't valid.
- None is the default value, which will redrive the whole queue
- get_messages was changed to not allow 0 as limit and raise an exception in case of limit lower than 1.

Kirill888 commented on libs/cloud/odc/aws/queue.py:
```diff
@@ -27,8 +27,10 @@ def post_messages(to_queue, messages):
             message.delete()
         return []
 
+    if limit < 1:
```
at this point `limit=None` is a possibility, so you need to check for that as you can't compare `None < 1`.
```suggestion
    if limit is not None and limit < 1:
```

carioca-au replied:
You are totally right. Also, limit is just used on **get_messages** which already is checking for that, so this check is useless.
Removed.
Thank you

Commit:
Limit 0 not allowed for redrive to queue:

- Limit 0 or lower, now raises an exception informing that value isn't valid.
- None is the default value, which will redrive the whole queue
- get_messages was changed to not allow 0 as limit and raise an exception in case of limit lower than 1.
